test/conftest.py

```diff
@@ -14,6 +19,10 @@
     CodeSegment,
 )
 from sqlfluff.core.templaters import TemplatedFile
+from test.dialects.parse_fixtures import load_file
+
+# When writing YAML files, double quotes string values needing escapes.
+oyaml.add_representer(str, quoted_presenter)
```

Comment:
Moved this code here from `test/generate_parse_fixture_yml.py` to ensure consistent YAML quoting format between that script and the test (`test/dialects/dialects_test.py::test__dialect__base_parse_struct`) that verifies those files.